### ISSUE TITLE
Code Insights: Keep collection routes plural

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#45225]
Follow up for https://github.com/sourcegraph/sourcegraph/pull/45004

### Plural collection routes 

**TL;DR** Currently we have following structure for for the code insight entity routes 
`/insights/insight/:id`, instead this we would like to have simplified version `/insights/:id` or more generic - `<collection name in plural form>/:entityId`. See this thread for more details. https://github.com/sourcegraph/sourcegraph/pull/45004#discussion_r1038465474

From @felixfbecker about collection-like routes 
>...the collection name should remain plural. I.e. a specific insights lies under `./:id` from the collection, and the collection lies at `..` from the insight. Think about this like folders in a file system, the folder is insights and the insights are all children of it. Similarly, dashboards is a folder and the dashboards are children of it.

### Edit-like routes

**TL;DR** Currently we have following structure for for the code insight entity routes 
`/insights/edit/:id`, instead this we would like to have `/insights/:id/edit` or more generic - `<collection name in plural form>/:entityId/edit`. See this thread for more details. https://github.com/sourcegraph/sourcegraph/pull/45004#discussion_r1038468764

> This was also the case before, but in general I think it would make more sense to have /insights/:id/edit rather than /edit/:id. The edit page is a specific view of the insight, it make less sense to have a top-level edit folder that contains all the edit views and mirrors the structure otherwise. ...
